[mobile][quantization] Add quantized torch mean implementation

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):

* **#27675 [mobile][quantization] Add quantized torch mean implementation**


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17989336](https://our.internmc.facebook.com/intern/diff/D17989336)

supriyar committed:
[mobile][quantization] Add quantized torch mean implementation
Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[mobile][quantization] Add quantized torch mean implementation"


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[mobile][quantization] Add quantized torch mean implementation"


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ljk53 commented on test/test_quantized.py:
```diff
@@ -886,19 +886,21 @@ def test_cat_nhwc(self, X, relu):
-        qY = torch.mean(qX, dim)
-
-        self.assertEqual(Y, qY.dequantize())
+    @given(X=hu.tensor(shapes=hu.array_shapes(min_dims=4, max_dims=4,
```
why do we change instead of adding new test cases? was it only for qnnpack before?

supriyar replied:
We want to eventually unify tests for qnnpack and server side. For cases like mean is it pretty straightforward to merge them.

supriyar committed:
Update on "[mobile][quantization] Add quantized torch mean implementation"


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[mobile][quantization] Add quantized torch mean implementation"

Stack from [ghstack](https://github.com/ezyang/ghstack):

* **#27675 [mobile][quantization] Add quantized torch mean implementation**


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17989336](https://our.internmc.facebook.com/intern/diff/D17989336)

[ghstack-poisoned]

supriyar committed:
Update on "[mobile][quantization] Add quantized torch mean implementation"

Stack from [ghstack](https://github.com/ezyang/ghstack):

* **#27675 [mobile][quantization] Add quantized torch mean implementation**


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17989336](https://our.internmc.facebook.com/intern/diff/D17989336)

[ghstack-poisoned]

supriyar committed:
Update on "[mobile][quantization] Add quantized torch mean implementation"

Stack from [ghstack](https://github.com/ezyang/ghstack):

* **#27675 [mobile][quantization] Add quantized torch mean implementation**


Summary:
This leverages QNNPACK global average pooling to perform torch.mean on input feature maps
Currently can only support mean along HxW plane in NCHW tensor.

Test Plan:
python test/test_quantized.py TestQuantizedOps.test_mean

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17989336](https://our.internmc.facebook.com/intern/diff/D17989336)

[ghstack-poisoned]
